[Console] Update console readme

mibragimov opened the pull request:
### Summary

This PR updates the Console readme file with information about the architecture changes, new features, and outstanding issues.

elasticmachine commented:
Pinging @elastic/platform-deployment-management (Team:Deployment Management)

alisonelizabeth commented on src/plugins/console/README.md:
```diff
@@ -234,3 +234,43 @@ Some autocomplete definitions need to be configured with dynamic values that can
 A list of variables is defined in the  `parametrizedComponentFactories` function in [`kb.js`](https://github.com/elastic/kibana/blob/main/src/plugins/console/public/lib/kb/kb.js) file. The values of these variables are assigned dynamically for every cluster. 
 Use these variables with curly braces, for example `{indices}`, `{types}`, `{id}`, `{username}`, `{template}`, `{nodes}` etc.
 
+### New features
```
[nit] I think we can probably move these as part of the `##Features` section that already exists above in the readme.

alisonelizabeth commented on src/plugins/console/README.md:
```diff
@@ -234,3 +234,43 @@ Some autocomplete definitions need to be configured with dynamic values that can
 Use these variables with curly braces, for example `{indices}`, `{types}`, `{id}`, `{username}`, `{template}`, `{nodes}` etc.
 
+### New features
+#### Kibana API support
```
Can you include a link to the PR where this was implemented for reference?

alisonelizabeth commented on src/plugins/console/README.md:
```diff
@@ -234,3 +234,43 @@ Some autocomplete definitions need to be configured with dynamic values that can
+Another change is using core http client instead of jquery to send requests to Elasticsearch and Kibana APIs. The core http client is used in the [`es.ts`](https://github.com/elastic/kibana/blob/main/src/plugins/console/public/lib/es/es.ts#L36) and [`autocomplete.ts`](https://github.com/elastic/kibana/blob/main/src/plugins/console/public/services/autocomplete.ts#L72) files.
+
+### Outstanding issues
+#### Autocomplete suggestions for Kinaba API endpoints
```
```suggestion
#### Autocomplete suggestions for Kibana API endpoints
```

alisonelizabeth commented on src/plugins/console/README.md:
````diff
@@ -234,3 +234,43 @@ Some autocomplete definitions need to be configured with dynamic values that can
+```
+
+### Architecture changes
+One of the main changes in architecture is refactoring the retrieval of autocomplete suggestions. Console used to send a separate request to ES for each autocomplete entity (mappings, aliases, templates, data-streams etc). This has been refactored to send a single request to Kibana API and retrieve all the autocomplete entities in one response. The autocomplete entities' data and methods are enca ...
````
Can you provide more background around why this decision was made, and link to any relevant PRs?

alisonelizabeth commented on src/plugins/console/README.md:
```diff
@@ -234,3 +234,43 @@ Some autocomplete definitions need to be configured with dynamic values that can
+### Architecture changes
+One of the main changes in architecture is refactoring the retrieval of autocomplete suggestions. Console used to send a separate request to ES for each autocomplete entity (mappings, aliases, templates, data-streams etc). This has been refactored to send a single request to Kibana API and retrieve all the autocomplete entities in one response. The autocomplete entities' data and methods are enca ...
+
+Another change is using core http client instead of jquery to send requests to Elasticsearch and Kibana APIs. The core http client is used in the [`es.ts`](https://github.com/elastic/kibana/blob/main/src/plugins/console/public/lib/es/es.ts#L36) and [`autocomplete.ts`](https://github.com/elastic/kibana/blob/main/src/plugins/console/public/services/autocomplete.ts#L72) files.
```
Same here - if you can provide more background around this change, I think that would be helpful.

alisonelizabeth reviewed (commented):
Thanks so much for working on this @mibragimov! Looks great. Left a few comments. I think wherever possible, if there's a related PR for something you discuss, it'd be helpful to link to it as a reference.

alisonelizabeth reviewed (approved):
Thanks for addressing my feedback @mibragimov! Latest LGTM.

alisonelizabeth commented:
@elasticmachine merge upstream

sabarasaba commented on src/plugins/console/README.md:
```diff
@@ -234,3 +260,22 @@ Some autocomplete definitions need to be configured with dynamic values that can
+- It added unnecessary dependencies on the ES proxy
+- We risked fetching an enormous response payload for users who have a large number of mappings.
+
+To address these issues, we created a specific [route](https://github.com/elastic/kibana/blob/main/src/plugins/console/server/routes/api/console/autocomplete_entities/index.ts) in the Kibana server to retrieve all autocomplete entities in one request and send it to the client. For more details, see [#130633](https://github.com/elastic/kibana/pull/130633).
```
Is it worthy to mention that this  response is capped to 10mb?

sabarasaba reviewed (approved):
changes lgtm!

ElenaStoeva reviewed (approved):
Thanks for doing this @mibragimov, that's very useful! LGTM

yuliacech reviewed (approved):
Thanks a lot for updating the readme, @mibragimov! 
LGTM 👍  just left one minor comment below

mibragimov commented:
Thanks, everyone for your reviews! 🙏 Addressed the comments with https://github.com/elastic/kibana/pull/149088/commits/bf261fee002d0b770ec894c4eb25fe6085d95796
